src/components/course/data/hooks.jsx

```diff
@@ -638,3 +558,103 @@ export const useCheckAccessPolicyRedemptionEligibility = ({
     queryFn: checkRedemptionEligiblity,
   });
 };
+
+export const useUserSubsidyApplicableToCourse = ({
+  courseData,
+  accessPolicyRedemptionEligibilityData,
+  isPolicyRedemptionEnabled,
+  subscriptionLicense,
+  courseService,
+  couponCodes,
+  canEnrollWithEnterpriseOffers,
+  enterpriseOffers,
+  onValidateSubscriptionLicenseForCourseError,
+}) => {
+  const courseMetadata = useMemo(() => {
+    if (!courseData) {
+      return undefined;
+    }
+    const {
+      catalog: {
+        containsContentItems,
+        catalogList: catalogsWithCourse,
+      },
+      courseDetails,
+    } = courseData;
+
+    return {
+      containsContentItems,
+      catalogsWithCourse,
+      courseDetails,
+    };
+  }, [courseData]);
+
+  const userSubsidyApplicableToCourse = useMemo(() => {
+    const validateSubscriptionLicenseForCourse = async () => {
+      const fetchLicenseSubsidyResponse = await courseService.fetchUserLicenseSubsidy();
+      return camelCaseObject(fetchLicenseSubsidyResponse.data);
+    };
+
+    if (!courseMetadata) {
+      return undefined;
+    }
+
+    // If course can be redeemed with EMET system, return the redeemable subsidy
+    // access policy.
+    if (isPolicyRedemptionEnabled) {
+      const firstRedeemableCourseRun = accessPolicyRedemptionEligibilityData.find(policy => policy.canRedeem);
+      const redeemablePolicyForCourseRun = firstRedeemableCourseRun.redeemableSubsidyAccessPolicy;
+      return {
+        discountType: 'percentage',
+        discountValue: 100,
+        subsidyType: LEARNER_CREDIT_SUBSIDY_TYPE,
+        perLearnerEnrollmentLimit: redeemablePolicyForCourseRun.perLearnerEnrollmentLimit,
+        perLearnerSpendLimit: redeemablePolicyForCourseRun.perLearnerSpendLimit,
+        policyRedemptionUrl: redeemablePolicyForCourseRun.policyRedemptionUrl,
+      };
```

Comment:
Matches structure used by all subsidy types. Transformed the redeemable subsidy access policy to conform to this existing structure to minimize downstream impacts in components that rely on `userSubsidyApplicableToCourse`.